cnd/src/btsieve/bitcoin/bitcoin_cache.rs

```diff
@@ -0,0 +1,94 @@
+use crate::btsieve::{BlockByHash, LatestBlock};
+use bitcoin::{hashes::sha256d, util::hash::BitcoinHash};
+use fmt::Debug;
+use futures::Future;
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use std::{fmt, fmt::Formatter, sync::Arc};
+use tokio::sync::Mutex;
+
+pub struct BitcoinCache<T> {
+    pub inner: T,
+    pub cache: Arc<Mutex<lru::LruCache<sha256d::Hash, bitcoin::Block>>>,
+}
+
+impl<T> Clone for BitcoinCache<T>
```

Comment:
I'm pretty sure we can just derive Clone for this.  @thomaseizinger any reason not too?  One reason may be to explicitly use
```
cache: Arc::clone(&self.cache),
```
But I do not see any real value in doing so.

Reply:
Now that we are back to one type parameter, it can be derived yes.